apps/wpcom-block-editor/src/wpcom/features/tracking.js

```diff
@@ -21,14 +21,28 @@ const debug = debugFactory( 'wpcom-block-editor:tracking' );
 /**
  * Looks up the block name based on its id.
  *
- * @param {string} blockId Blog identifier.
- * @returns {string|null} Blg name if it exists. Otherwise, `null`.
+ * @param {string} blockId Block identifier.
+ * @returns {string|null} Block name if it exists. Otherwise, `null`.
  */
 const getTypeForBlockId = blockId => {
 	const block = select( 'core/block-editor' ).getBlock( blockId );
 	return block ? block.name : null;
 };
 
+/**
+ * Ensure you are working with block object. This either returns the object
+ * or tries to lookup the block by id.
+ *
+ * @param {string|object} block Block object or string identifier.
+ * @returns {object|null} block object or null.
+ */
+const ensureBlockObject = block => {
+	if ( typeof block === 'object' ) {
+		return block;
+	}
+	return select( 'core/block-editor' ).getBlock( block );
```

Comment:
do you know if it could return something different than an object? If so maybe we should return an empty object instead.

Reply:
It can return `null` as documented. Not sure if we should account for that in `trackBlocksHandler()`?

Reply:
it should return always an object if I'm not wrong because this value will be used by the `propertiesHandler` callback...

```es6
   ...propertiesHandler( block, parentBlock ),
```

which usually destructs this object to get some block attributes, such as the `name`

Reply:
Yeah I can get on board with that, the function _is_ called `ensureBlockObject` after all

Reply:
```suggestion
	const block = select( 'core/block-editor' ).getBlock( block );
	if ( typeof block !== 'object' ) {
		return {};
	}
	
	return block;
```

Reply:
now returning an empty object